### PR TITLE
Update beat developer guide with new interface. Closes #330

### LINKED ISSUE #330: Update Developer guide for Interface changes
See:
* https://github.com/elastic/libbeat/pull/315
* https://github.com/elastic/topbeat/pull/141
* https://github.com/elastic/filebeat/pull/290
* https://github.com/elastic/packetbeat/pull/401

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

**Comment by ruflin** [comment]
@dedemorton Can you review this?

File: docs/newbeat.asciidoc [hunk 1]
```diff
@@ -111,22 +111,43 @@ This means your Beat should implement the following methods:
...
+For more complex beats, the optional FlagsHandler interface is available.
+
+[source,go]
+----------------------------------------------------------------------
+type FlagsHandler interface {
+	HandleFlags(*Beat)
+}
+----------------------------------------------------------------------
+
+The FlagsHandler Beater extension is for handling flags input on startup.
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: This read a bit awkwardly for me the first couple of times. I'd suggest moving up the statement you make later on because I think it's easier to follow. So replace this sentence with something like: "You can use the FlagsHandler interface to add additional command line flags to your Beat." (and then remove the sentence where you say this later).

File: docs/newbeat.asciidoc [hunk 2]
```diff
@@ -111,22 +111,43 @@ This means your Beat should implement the following methods:
...
+The HandleFlags callback will be called after parsing the command line
+arguments and handling the '--help' or '--version' flags. Use this interface
+to add additional command line flags to your beat. For an example on how to
+implement HandleFlags, have a look at
+https://github.com/elastic/packetbeat/blob/master/beat/packetbeat.go[packetbeat]
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: capitalize Packetbeat here and add a period to the end.

File: docs/newbeat.asciidoc [hunk 3]
```diff
@@ -111,22 +111,43 @@ This means your Beat should implement the following methods:
...
+
+[source,go]
+----------------------------------------------------------------------
+type FlagsHandler interface {
+	HandleFlags(*Beat)
+}
+----------------------------------------------------------------------
+
+The FlagsHandler Beater extension is for handling flags input on startup.
+The HandleFlags callback will be called after parsing the command line
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: I think we need to make it clear here that we're talking about the command line flags inherited from libbeat. Maybe try this: "The HandleFlags callback is called after the Beat parses any command line arguments inherited from libbeat and handles the `--help` and `--version` flags." (Make "command line arguments" a link to command-line-options.html, assuming those are the options that get parsed first. I'm not sure if I've nailed the language here, but hopefully you understand my concern. I wonder if you could even drop "...and handles the --help and --version flags" since those are also inherited from libbeat).

File: docs/newbeat.asciidoc [hunk 4]
```diff
@@ -111,22 +111,43 @@ This means your Beat should implement the following methods:
...
+arguments and handling the '--help' or '--version' flags. Use this interface
+to add additional command line flags to your beat. For an example on how to
+implement HandleFlags, have a look at
+https://github.com/elastic/packetbeat/blob/master/beat/packetbeat.go[packetbeat]
+
+We strongly recommend to only put the main method into the main package.
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: The way the sentence is worded currently it sounds like maybe the user could put the method in a different package. But I think what you're saying is that you want the main package to contain the main method only. Maybe this change would help: "We strongly recommend that the main package contain the main method only."

File: docs/newbeat.asciidoc [hunk 5]
```diff
@@ -111,22 +111,43 @@ This means your Beat should implement the following methods:
...
+This will allow all other parts of your beat to be used as library by
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: Missing article. Need to say one of the following: "...to be used as a library" or "...to be used as libraries"

File: docs/newbeat.asciidoc [hunk 6]
```diff
@@ -111,22 +111,43 @@ This means your Beat should implement the following methods:
...
+other beats if needed. This means all your beat specific code goes for example
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: Maybe reword this slightly and say, "So, for example, you could put all your beat-specific code in a separate beat folder and package." It just sounds a little more optional when you put it this way.;

File: docs/newbeat.asciidoc [hunk 7]
```diff
@@ -308,29 +329,22 @@ func (t *Topbeat) Stop() {
...
-	b.Run()
+var Version = "1.0.0"
+var Name = "topbeat"
 
+func main() {
+	beat.Run(Name, Version, topbeat.New())
 }
 ----------------------------------------------------------------------
 
+We recommend to implement a New function that creates your beats object.
```
Comment (dedemorton) [inline docs/newbeat.asciidoc]: Shouldn't we show some example code for the New function? Does the New function call the CommandLineSetup, LoadConfig, and Config methods shown in the code example that you've deleted? If not, where does those methods get called?

Comment (dedemorton) [inline docs/newbeat.asciidoc]: Edit: Change beats object to Beats object.

Comment (dedemorton) [inline docs/newbeat.asciidoc]: @ruflin I am done with my review.

**Comment by ruflin** [comment]
@dedemorton does it perhaps make more sense that I give you access to my fork? Like this you can directly edit the text on Github in my PR or push to my PR?

**Comment by dedemorton** [comment]
OK, I've added my edits, but I made some additional changes that I didn't propose during my review, so please check them again.

**Comment by andrewkroh** [comment]
LGTM
